api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -213,49 +215,46 @@ private CodeableConcept getManifestation(List<AllergyReaction> reactions) {
 	}
 	
 	private AllergyIntolerance.AllergyIntoleranceSeverity getFhirSeverity(Concept severityConcept) {
-		if (severityConcept == null) {
+		getSeverityConceptUuids();
+		
+		if (severityConcept == null || severityConceptUuids.size() < 4) {
 			return null;
 		}
 		
-		if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MILD, "")
-		        .equals(severityConcept.getUuid())) {
+		if (severityConcept.getUuid().equals(severityConceptUuids.get(0))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.MILD;
-		} else if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_MODERATE, "")
-		        .equals(severityConcept.getUuid())) {
+		} else if (severityConcept.getUuid().equals(severityConceptUuids.get(1))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.MODERATE;
-		} else if (globalPropertyService.getGlobalProperty(FhirConstants.GLOBAL_PROPERTY_SEVERE, "")
-		        .equals(severityConcept.getUuid())) {
+		} else if (severityConcept.getUuid().equals(severityConceptUuids.get(2))) {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.SEVERE;
 		} else {
 			return AllergyIntolerance.AllergyIntoleranceSeverity.NULL;
 		}
 	}
 	
 	private Concept getOpenmrsSeverity(AllergyIntolerance.AllergyIntoleranceSeverity severity) {
-		Concept concept;
+		getSeverityConceptUuids();
+		
+		if (severityConceptUuids.size() < 4) {
+			return null;
+		}
```

Comment:
What if ` severityConceptUuids.size() == 3`. This will return null which should not be the case. It should still return switch for the 3 UUIDs in the array and return there the concept